### PR TITLE
[WIP] Schema: Add `Folder` kind

### PR DESCRIPTION
**Pull request by undef1nd**
**Special notes for your reviewer**:
This is not the final definition of the schema, but rather a first step. It's a union of the fields of the following existing types:
[types/folders.ts](https://github.com/grafana/grafana/blob/main/public/app/types/folders.ts#L5-L15),
[services/folder/model.go](https://github.com/grafana/grafana/blob/main/pkg/services/folder/model.go#L26-L44),
[api/dtos/folder.go](https://github.com/grafana/grafana/blob/main/pkg/api/dtos/folder.go#L9-L27),
[pkg/models/folders.go](https://github.com/grafana/grafana/blob/main/pkg/models/folders.go#L10-L23)

### CODE REVIEW

File: kinds/structured/folder/folder_kind.cue [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+		schemas: [
+			{//0.0
+				// OrgId is the ID of an organisation the folder belongs to.
+				orgId: int64 @grafanamaturity(ToMetadata="sys")
+				// Uid is the unique numeric identifier for the folder.
+				uid: string
+				// Title of the folder.
+				title: string
+				// TODO
+				url: string
```
Comment (sdboyer) [inline kinds/structured/folder/folder_kind.cue:17]: probably optional?

Comment (undef1nd) [reply kinds/structured/folder/folder_kind.cue:17]: `url`? Not sure, gonna take a look how it's used in the code.

Comment (ryantxu) [reply kinds/structured/folder/folder_kind.cue:17]: Are we trying to replicate/replace the existing tangled mess?

Comment (sdboyer) [reply kinds/structured/folder/folder_kind.cue:17]: We're following the [maximalist field selection rule](https://docs.google.com/document/d/1gPzSCQn-0Qg86PYpPcR-l_kTQ4Ba1wgnNLZFKPSe8yQ/edit#heading=h.jrv2hv50f2cu), per [the WIP guide](https://docs.google.com/document/d/1X_R-R5LIhGoqvZUec7tQyaPXduQW_qivWJjFao5-LJ4/edit#heading=h.am98mp3y9asd) procedure for arriving at `merged` maturity.

The [`@grafanamaturity` attributes](https://docs.google.com/document/d/1gPzSCQn-0Qg86PYpPcR-l_kTQ4Ba1wgnNLZFKPSe8yQ/edit#heading=h.iueyrgykhe1k) indicate the plan for how most of the fields are likely to go away. The remainder can be discussed in a follow-up PR.

Comparing to what you linked, i suppose we should add a `@grafanamaturity(MaybeRemove)` to `url`, here? Or if it's supposed to be a computed value, it should be `@grafanamaturity(ToMetadata="kind")` according to what we outlined in #59399

File: kinds/structured/folder/folder_kind.cue [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+				// Create is the date when folder was created.
+				created: string @grafanamaturity(ToMetadata="sys")
+				// UpdatedBy is the ID of a user who modified the folder most recently.
+				updatedBy: int64 @grafanamaturity(ToMetadata="sys")
+				// Update is the date of the most recent folder update.
+				updated: string @grafanamaturity(ToMetadata="sys")
+				// Version of the folder, incremented each time it is updated.
+				version: int64 @grafanamaturity(ToMetadata="kind", MaybeRemove)
+				// AccessControl metadata associated with a given resource.
+				accessControl: [string]: bool @grafanamaturity(ToMetadata="sys")
```
Comment (Clarity-89) [inline kinds/structured/folder/folder_kind.cue:43]: `accessControl` is currently optional on front end, so I think it'd be optional here as well. Or will it create any issues for the go types?

Comment (undef1nd) [reply kinds/structured/folder/folder_kind.cue:43]: There's an issue with generating map type as optional https://github.com/grafana/grafana/pull/60188#discussion_r1047130066

**Pull request closed without merging**